util/Logger.h

```diff
@@ -1,55 +1,245 @@
+#ifndef FREEORION_WIN32
+
+#define FO_GLOBAL_LOGGER_NAME(name) fo_logger_global_##name
+
+#else
+
+// Note: The is a workaround for Visual C++ non-conformant pre-processor
+// handling of empty macro arguments.
+// https://msdn.microsoft.com/en-us/library/hh567368.aspx
+// https://blogs.msdn.microsoft.com/vcblog/2017/03/07/c-standards-conformance-from-microsoft/
+#define FO_GLOBAL_LOGGER_NAME_NO_ARG() fo_logger_global_
+#define FO_GLOBAL_LOGGER_NAME_ONE_ARG(name) fo_logger_global_##name
+#define FO_GLOBAL_LOGGER_NAME(...)                                      \
+    BOOST_PP_IF(BOOST_PP_IS_EMPTY(__VA_ARGS__),                         \
+                FO_GLOBAL_LOGGER_NAME_NO_ARG(),                         \
+                FO_GLOBAL_LOGGER_NAME_ONE_ARG(__VA_ARGS__))
+
+#endif
 
 /** Initializes the logging system. Log to the given file.  If the file already
- * exists it will be deleted. \p root_logger_name is the name by which the
+ * exists it will be deleted. \p default_exec_logger_name is the name by which the
  * default logger "" appears in the log file.*/
-FO_COMMON_API void InitLoggingSystem(const std::string& logFile, const std::string& root_logger_name);
-
-/** Accessors for the App's logger */
-FO_COMMON_API void SetLoggerPriority(int priority);
-
+FO_COMMON_API void InitLoggingSystem(const std::string& logFile, const std::string& default_exec_logger_name);
+
+/** Overrides all logger thresholds to \p threshold.*/
+FO_COMMON_API void OverrideLoggerThresholds(const LogLevel threshold);
+
+FO_COMMON_API const std::string& DefaultExecLoggerName();
+
+/** A type for loggers (sources) that allows for severity and a logger name (channel in
+    boost parlance) and supports multithreading.*/
+using NamedThreadedLogger = boost::log::sources::severity_channel_logger_mt<
+    LogLevel,     ///< the type of the severity level
+    std::string   ///< the channel name of the logger
+    >;
+
+// Setup file sink, formatting, and \p name channel filter for \p logger.
+FO_COMMON_API void ConfigureLogger(NamedThreadedLogger& logger, const std::string& name);
+
+// Place in source file to create the previously defined global logger \p name
+#define CreateThreadedLogger(name)                                      \
+    BOOST_LOG_INLINE_GLOBAL_LOGGER_INIT(                                \
+        FO_GLOBAL_LOGGER_NAME(name), NamedThreadedLogger)               \
+    {                                                                   \
+        auto lg = NamedThreadedLogger(                                  \
+            (boost::log::keywords::severity = LogLevel::debug),         \
+            (boost::log::keywords::channel = #name));                   \
+        ConfigureLogger(lg, #name);                                     \
+        return lg;                                                      \
+    }
+
+// Signal that logger \p name has been created
+using LoggerCreatedSignalType = boost::signals2::signal<void (const std::string logger)>;
```

Comment:
No, the signal may cross threads and then the reference results in undefined behavior.